src/lib/ColumnToCode.php

```diff
@@ -314,10 +327,18 @@ private function resolveDefaultValue():void
     private function isDefaultAllowed():bool
     {
         $type = strtolower($this->column->dbType);
-        if ($type === 'tsvector') {
-            return false;
+        switch ($type){
+            case 'tsvector':
+                return false;
+            case 'blob':
+            case 'geometry':
+            case 'text':
+            case 'json':
+                return ($this->isMysql() && !$this->isMariaDb()) === false;
```

Comment:
condition looks hard to read, should be positive check instead of negative.

Reply:
Condition:

It was like this before: (refactored in this PR to use switch-case)

```php
return !($this->isMysql() && !$this->isMariaDb() && in_array($type, ['blob', 'geometry', 'text', 'json']));
```

  | is mysql | is mariadb | type in array | final | global negation
-- | -- | -- | -- | -- | --
mysql json | T | !F=T | T | T | F
mysql nonjson | T | !F=T | F | F | T
mariadb json | F | !T=F | T | F | T
mariadb nonjson | F | !T=F | F | F | T

---

Default are not allowed for  BLOB, TEXT, GEOMETRY, and JSON in MySQL [prior to MySQL 8.0.13](https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html)

Curious to know why it is allowed for MariaDB?

Reply:
mariadb allows default value for JSON column at least, have been using that a few times. e.g. `data JSON NOT NULL DEFAULT '{}'`